### PR TITLE
Add a couple of notes around gotchas with the Client Dependency Framework

### PR DESCRIPTION
**Pull request by marcemarc**
see https://github.com/umbraco/UmbracoDocs/issues/3076

### CODE REVIEW

File: Getting-Started/Design/Stylesheets-Javascript/index.md [hunk 1]
```diff
@@ -102,4 +102,13 @@ You can bundle and minify as follows in a view template file.
 </head>
 ```
 
+:::note
+If you are running your site locally and in your Web.Config the compilation setting for debug is set to true - the Client Dependency Framework will not minify or combine your js/css. This is by design to help when debugging issues in a local environment - but also means if there is a problem minifying a file - you'll only encounter the issue when you deploy - so do test minification and bundling locally by setting the debug setting to false for compilation.
```
Comment (sofietoft) [inline Getting-Started/Design/Stylesheets-Javascript/index.md:106]:
```suggestion
If you are running your site locally and the compilation setting for debugging in your `web.config` is set to true, the Client Dependency Framework will not minify or combine your js/CSS. This is by design to help when debugging issues in a local environment. However, it also means if there is a problem minifying a file you will only encounter the issue when you deploy. To test minification and bundling locally, set the debug setting to false for compilation.
```

File: Getting-Started/Design/Stylesheets-Javascript/index.md [hunk 2]
```diff
@@ -102,4 +102,13 @@ You can bundle and minify as follows in a view template file.
...
+:::
+
+:::note
+When adding stylesheet references to Html.RequiresCss there is no need to add ?v=1 to specify a specific version of the asset for cachebusting (In fact doing so will break the minification and combination of your files!). The Client Dependency Framework uses the version number from /config/clientdependency.config on the querystring of the combined and minified resource. To force your updated minified and combined asset to be refreshed, increment this version number by 1 in the config file to bust any browser caching on the old version of your assets. See [Healthcheck for Client Dependency Framework](https://our.umbraco.com/packages/developer-tools/health-check-for-client-dependency-framework/) package to do this from the backoffice, (or automatically increment the version in your build and deploy process).
```
Comment (sofietoft) [inline Getting-Started/Design/Stylesheets-Javascript/index.md:110]:
```suggestion
When adding stylesheet references to `Html.RequiresCss` there is no need to add `?v=1` to specify a specific version of the asset for cache-busting. Doing so will break the minification and combination of your files. The Client Dependency Framework uses the version number from `/config/clientdependency.config` on the querystring of the combined and minified resource. 

To force your updated, minified and combined assets to be refreshed, increment this version number by 1 in the config file to bust any browser caching on the old version of your assets. See the [Healthcheck for Client Dependency Framework](https://our.umbraco.com/packages/developer-tools/health-check-for-client-dependency-framework/) package to do this from the backoffice, or you can automatically increment the version in your build and deploy process. 
```

**Review by sofietoft — COMMENTED** [review]
Thanks for adding these notes @marcemarc ! 💪 

I've made a few suggestions here and there - mostly to eliminate those looong sentences 😄 

Also, any reason it's split into two notes, instead of keeping it in one? 🤔 Just a though...

**Comment by marcemarc** [comment]
@sofietoft ace thanks! I did sort of dash it off without a care in the world... :-P

I think it's worth having as two different notes, because they are two very different gotchas... depending on what we think should happen over the external resource, it might make sense to convert from notes, into a 'Troubleshooting' section in it's own right...

**Review by sofietoft — APPROVED** [review]
Good point about why they are two separate notes! Let's leave it like that then 💪

**Comment by sofietoft** [comment]
Quick question @marcemarc - Would merging this PR close #3076 ? 🤞

**Comment by marcemarc** [comment]
@sofietoft not completely... it addresses the first two bullet points, merging it now would at least make that information available to people.

I've replied on the PR about the other points!

**Comment by sofietoft** [comment]
Ah, OK. Cool! 💪 
I'll merge this now then - but not the issue as of yet.

Thanks again! 😄

**Pull request merged**